assets/js/aplus.js

```diff
@@ -584,65 +588,182 @@ $.fn.highlightCode = function(options) {
     }
 
     $.extend(AplusTableFilter.prototype, {
+ 
+      init: function() {
+        var columnCount = 0;
+        this.element.find('thead').find('tr').each(function() {
+          var count = $(this).find('th').size();
+          columnCount = count > columnCount ? count : columnCount;
+        });
 
-        init: function() {
-          var columnCount = 0;
-          this.element.find('thead').find('tr').each(function() {
-            var count = $(this).find('th').size();
-            columnCount = count > columnCount ? count : columnCount;
-          });
+        var self = this;
+        var filterDelay = function(event) {
+          var input = $(this);
+          clearTimeout(self.timeout);
+          self.timeout = setTimeout(function() {
+            self.filterColumn(input);
+          }, 500);
+        };
+
+        this.filters = [];
+        var filterRow = $('<tr class="tableexport-ignore"></tr>');
+        for (var i = 0; i < columnCount; i++) {
+          this.filters.push('');
+          var filterInput = $('<input type="text" data-column="'+i+'">')
+            .on('keyup', filterDelay).on('change', filterDelay);
+
+
+          var filterCell = $('<td data-toggle="tooltip" title="' + currentScript.attr("filterHelperTrans") + '"></td>');
```

Comment:
This could use the JS translation api and thus have only `_("english test")` instead of using currentScript and django translations. A+ uses the translation api for usertags, so you can look how those are done. I think we could have two json files, "common" and "course staff".

Reply:
Why do we want two json files? Currently there is `teacher.fi.json` for translations.